Code-the-Dream-School/gg-pac-team3-monorepo
# Testing part 2

I made important updates to the courseController, lessonController, and their respective models to ensure that courseId and lessonId are generated by Firestore rather than being provided by the frontend. This change prevents confusion and potential issues during testing, where IDs were previously being set before creation. Now, IDs are correctly assigned immediately after a course or lesson is created.

**Testing Files:**

- courseController.tests.mjs: Tests for course creation, fetching, updating, and deletion.

- lessonController.tests.mjs: Tests for lesson creation, fetching, updating, and deletion.

**Testing Instructions:**

1. I updated the package.json to have the option to test each file.
2. Before running the tests make sure you have all dependencies.
3.  **To run all the tests:**
        `npm run dev `
       ` npm test`
4. **To do individual testings:**
    
    To run the user controller tests:
       `  npm run test:user`
     
     To run the frontpage tests:
         `npm run dev`
         `npm run test:frontpage`
      
      To run the course controller tests:
         `npm run test:course`
      
      To run the lesson controller tests:
          `npm run test:lesson`

<img width="578" alt="Screenshot 2024-08-25 at 4 41 36 PM" src="https://github.com/user-attachments/assets/d8cf0115-d620-4199-a1f5-30ec3c2a061b">
